File: ms-participants/src/main/java/com/emm/msparticipants/services/ParticipantService.java
package com.emm.msparticipants.services;

import com.emm.msparticipants.dtos.CreateParticipantDTO;
import com.emm.msparticipants.dtos.ParticipantDTO;

import java.util.List;

public interface ParticipantService {
    ParticipantDTO createParticipant(CreateParticipantDTO dto);
    List<ParticipantDTO> getAllParticipants();
    ParticipantDTO getParticipantById(Long id);
    void deleteParticipant(Long id);
    ParticipantDTO updateParticipant(Long id, CreateParticipantDTO dto);



}

File: matdash-free-angular-v3/matdash-free-angular-v3/main/src/app/providers/services/participant/participant.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {CreateParticipantDTO} from "../../../models/createParticipantDTO";
import {ParticipantDTO} from "../../../models/ParticipantDTO ";
import {environment} from "../../../../environments/environment";

@Injectable({
  providedIn: 'root'
})
export class ParticipantService {

  private readonly baseUrl =  `${environment.url}participants`;

  constructor(private http: HttpClient) {}

  createParticipant(dto: CreateParticipantDTO): Observable<ParticipantDTO> {
    return this.http.post<ParticipantDTO>(this.baseUrl, dto);
  }

  getAllParticipants(): Observable<ParticipantDTO[]> {
    return this.http.get<ParticipantDTO[]>(this.baseUrl);
  }

  getParticipantById(id: number): Observable<ParticipantDTO> {
    return this.http.get<ParticipantDTO>(`${this.baseUrl}/${id}`);
  }

  updateParticipant(id: number, dto: CreateParticipantDTO): Observable<ParticipantDTO> {
    return this.http.put<ParticipantDTO>(`${this.baseUrl}/${id}`, dto);
  }

  deleteParticipant(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }
}
